### ISSUE TITLE
Trac-preproc seems to fail when nrun is 1 

### ISSUE THREAD

**Issue by vinferrer** [freesurfer/freesurfer#929]
Hello all, 

I was trying trac-all -prep in one of my subjects and it was giving me an error at the eddycorrection `eddy_openmp`, I realised that the error comes from a previous problem. If we follow the output from trac-all at this point:
```
writing to /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.1.nii.gz...
INFO: found /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig.1.bvals, copying
INFO: found /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig.1.bvecs, converting to LAS
mv -f /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.1.nii.gz /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.nii.gz
mv -f /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.1.bvecs /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.bvecs
mv -f /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.1.bvals /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.bvals
mri_convert --frame 0 /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.nii.gz /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/lowb_orig_las.nii.gz
mri_convert --frame 0 /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.nii.gz /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/lowb_orig_las.nii.gz
reading from /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.nii.gz...
TR=900.00, TE=0.00, TI=0.00, flip angle=0.00
i_ras = (-1, 1e-16, 0)
j_ras = (1e-[16, 0.92521, 0.379456](callto:16, 0.92521, 0.379456))
k_ras = (-3.79456e-[17, -0.379456, 0.92521](callto:17, -0.379456, 0.92521))
keeping frame 0
writing to /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/lowb_orig_las.nii.gz...
bet /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/lowb_orig_las.nii.gz /bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/lowb_orig_las_brain.nii.gz -m -f 0.3
awk: fatal: cannot open file `/bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.1.bvals' for reading (No such file or directory)
eddy_openmp --imain=/bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.nii.gz --mask=/bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/lowb_orig_las_brain_mask.nii.gz --bvecs=/bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.bvecs --bvals=/bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi_orig_las.bvals --acqp=/bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/acqp.txt --index=/bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/index.txt --data_is_shelled --out=/bcbl/home/public/PD_ANALYSIS/FREESURFER_PPMI/sub-3102ses-Session_baseline_final/dmri/dwi
EddyInputError:  Error when attempting to read --index file
Terminating program
Linux ipsnode-7-3.local [3.10.0-693.21.1](callto:3.10.0-693.21.1).el7.x86_64 #1 SMP Wed Mar 7 19:03:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

trac-preproc exited with ERRORS at Fri Feb 11 09:37:18 CET 2022
```
You can see that the error comes from the --index file. However this is due to the file being empty. The file is empty because the script is unable to read the file `dwi_orig_las.1.bvals` in line 17 of this error. This is happening because since there is only one run this file is renamed to `dwi_orig_las.bvals` in line 5. 

I have a quick fix to this problem I would like to submit as a PR however I considered an issue was better as a first approach to report the error. My idea would be to add a if condition around line:
https://github.com/freesurfer/freesurfer/blob/68395e7d1cf35b9666e9094bff2a96fc0587c5a2/scripts/trac-preproc#L698
something as:
```
    @ irun = 1
    if ($nrun == 1) then
    	set indlist = ($indlist \
        `awk -v irun=$irun '{printf irun" "}' $dwibase.bvals`)
    else
        while ($irun <= $nrun)		# For each input DWI run
          set indlist = ($indlist \
           `awk -v irun=$irun '{printf irun" "}' $dwibase.$irun.bvals`)

          @ irun = $irun + 1
        end
    endif	
```
This way the behavior for multiruns is preserved and when only 1 run is present the code doesn't fail. If you agree to it, I can open a PR to incorporate it

**Comment by ayendiki** [comment]
Hi Vicente, any chance that this is the same as this previously reported and fixed issue?
https://www.mail-archive.com/freesurfer@nmr.mgh.harvard.edu/msg71427.html

That fix was merged into the dev version, and will be included in the 7.3 version.

**Comment by vinferrer** [comment]
It seems to be the case, I wasn't aware of this, we can close the issue then.
